app/models/nic/interface.rb

```diff
@@ -46,7 +54,32 @@ def normalize_ip
       self.ip = Net::Validations.normalize_ip(ip)
     end
 
+    # ensure that host name is fqdn
+    # if the user inputted short name, the domain name will be appended
+    # this is done to ensure compatibility with puppet storeconfigs
     def normalize_name
+      # no hostname was given or a domain was selected, since this is before validation we need to ignore
+      # it and let the validations to produce an error
+      return if name.empty?
+
+      # Remove whitespace
+      self.name.gsub!(/\s/,'')
+
+      if domain.nil? and name.match(/\./)
+        # try to assign the domain automatically based on our existing domains from the host FQDN
+        self.domain = Domain.all.select{|d| name.match(d.name)}.first rescue nil
```

Comment:
At a guess, it's in case we get nil in the middle of the evaluation. @ares would `.try(:first)` be better here?